core/src/main/scala/org/apache/spark/scheduler/cluster/cook/CoarseCookSchedulerBackend.scala

```diff
@@ -115,20 +118,34 @@ private[spark] class CoarseCookSchedulerBackend(
   }
 
   var runningJobUUIDs = Set[UUID]()
+  var abortedJobIds = Set[UUID]()
+
+  private var jobLimitOption: Option[Int] = None
 
   val jobListener = new CJobListener {
     // These are called serially so don't need to worry about race conditions
     def onStatusUpdate(job : Job) {
-      if (job.getStatus == Job.Status.COMPLETED && !job.isSuccess) {
+      val uuid = job.getUUID
+      val isCompleted = job.getStatus == Job.Status.COMPLETED
+      val isAborted = abortedJobIds.contains(uuid)
+
+      if (isCompleted) {
         totalCoresRequested -= job.getCpus().toInt
-        totalFailures += 1
-        logWarning(s"Job ${job.getUUID} has died. Failure ($totalFailures/$maxFailures)")
-        runningJobUUIDs = runningJobUUIDs - job.getUUID
-        if (totalFailures >= maxFailures) {
-          // TODO should we abort the outstanding tasks now?
-          logError(s"We have exceeded our maximum failures ($maxFailures)" +
-                    "and will not relaunch any more tasks")
-        } else requestRemainingCores()
+        abortedJobIds = abortedJobIds - uuid
+        runningJobUUIDs = runningJobUUIDs - uuid
+
+        if (!job.isSuccess && !isAborted) {
```

Comment:
Might be nice to log something when we notice that an aborted job was successfully aborted, either at INFO or DEBUG level